contracts/GenArt721CoreV3.sol

```diff
@@ -1425,13 +1456,20 @@ contract GenArt721CoreV3 is
         onlyValidTokenId(_tokenId)
         returns (string memory)
     {
-        return
-            string(
-                abi.encodePacked(
-                    projects[_tokenId / ONE_MILLION].projectBaseURI,
-                    Strings.toString(_tokenId)
-                )
-            );
+        string memory _projectBaseURI = projects[_tokenId / ONE_MILLION]
+            .projectBaseURI;
+        uint256 _projectBaseURILen;
+        assembly {
+            // load string length directly from first word
+            _projectBaseURILen := mload(_projectBaseURI)
+        }
```

Comment:
Do we need the fancy-ness of doing this in assembly?

Reply:
This is typically called as a read-only, so yeah may as well cast as bytes and strip out the assembly. pushed an update!

Reply:
Fantastic–TYVM 🙏 🙏